File: Leger/PolyPaint/app/src/main/java/com/polypaint/polypaint/Model/BasicShape.kt
package com.polypaint.polypaint.Model

import android.content.Context
import com.polypaint.polypaint.R
import android.view.LayoutInflater
import java.io.Serializable
import java.util.*
import kotlin.collections.ArrayList


open class BasicShape (var id: String = "", var type: Int = 0, var name: String = "", var shapeStyle: ShapeStyle , var linksTo: ArrayList<String?> = ArrayList(), var linksFrom: ArrayList<String?> = ArrayList()) : Serializable, DrawingElement(id) {
//    private var inflater : LayoutInflater? = null

    init {

        //        inflater = getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

//        val mBasicElem = BasicElementView(this)

//        val viewToAdd = inflater!!.inflate(R.layout.basic_element, null)
//        mBasicElem.addView(viewToAdd,mBasicElem.childCount - 1)
    }
    fun copy(id: String = this.id, type: Int = this.type, name: String = this.name, shapeStyle: ShapeStyle = this.shapeStyle.copy(), linksTo: ArrayList<String?> = ArrayList(), linksFrom: ArrayList<String?> = ArrayList()) = BasicShape(id, type, name, shapeStyle, linksTo, linksFrom)

}

File: Lourd/PolyPaint/PolyPaint/Templates/BasicShape.cs
using PolyPaint.CustomInk;
using System.Collections.Generic;

namespace PolyPaint.Templates
{
    public class BasicShape
    {
        public string id { get; set; }
        public int type { get; set; }
        public string name { get; set; }
        public ShapeStyle shapeStyle { get; set; }
        public List<string> linksTo { get; set; }
        public List<string> linksFrom { get; set; }

        public BasicShape()
        {
        }

        public BasicShape(bool isFromServer, string id, int type, string name, ShapeStyle shapeStyle, List<string> linksTo, List<string> linksFrom)
        {
            this.id = id;
            this.type = type;
            this.name = name;
            this.shapeStyle = shapeStyle.Clone();

            if (isFromServer)
            {
                this.shapeStyle.coordinates.x /= CustomStroke.WIDTH;
                this.shapeStyle.coordinates.y /= CustomStroke.HEIGHT;
                this.shapeStyle.width /= CustomStroke.WIDTH;
                this.shapeStyle.height /= CustomStroke.HEIGHT;
            }
            else
            {
                this.shapeStyle.coordinates.x *= CustomStroke.WIDTH;
                this.shapeStyle.coordinates.y *= CustomStroke.HEIGHT;
                this.shapeStyle.width *= CustomStroke.WIDTH;
                this.shapeStyle.height *= CustomStroke.HEIGHT;
            }

            this.linksTo = linksTo;
            this.linksFrom = linksFrom;
        }

    }
}